cosmic/sample/sampler/cmc.py

```diff
@@ -114,57 +105,55 @@ def get_cmc_sampler(
     mass_binaries += np.sum(mass1_binaries)
     mass_binaries += np.sum(mass2_binaries)
 
+    # set singles id
+    single_ids = np.arange(mass1.size)
+    binary_secondary_object_id = np.arange(mass1.size, mass1.size + mass2_binaries.size)
+
+    # set binind and correct masses of binaries
+    binind = np.zeros(mass1.size)
+    binind[binary_index] = np.arange(len(binary_index)) + 1
+    mass1[binary_index] += mass2_binaries
+
     # track the total number sampled
     n_singles += len(mass_single)
     n_binaries += len(mass1_binaries)
 
     # select out the primaries and secondaries that will produce the final kstars
-    mass1_binary = np.array(mass1_binaries)
-    mass2_binary = np.array(mass2_binaries)
-    ecc = initconditions.sample_ecc(ecc_model, size=mass1_binary.size)
+    ecc = initconditions.sample_ecc(ecc_model, size=mass1_binaries.size)
+    porb_max = initconditions.calc_porb_max(mass1, vr, vt, binary_index, mass1_binaries, mass2_binaries, **kwargs)
     porb = initconditions.sample_porb(
-        mass1_binary, mass2_binary, ecc, porb_model, size=mass1_binary.size
+        mass1_binaries, mass2_binaries, ecc, porb_model, porb_max, size=mass1_binaries.size
     )
-    sep = utils.a_from_p(porb, mass1_binary, mass2_binary)
-    kstar1 = initconditions.set_kstar(mass1_binary)
-    kstar2 = initconditions.set_kstar(mass2_binary)
+    sep = utils.a_from_p(porb, mass1_binaries, mass2_binaries)
+    kstar1 = initconditions.set_kstar(mass1_binaries)
+    kstar2 = initconditions.set_kstar(mass2_binaries)
 
     # obtain radius
     Reff = initconditions.set_reff(mass1, metallicity=met, **kwargs)
     Reff1 = Reff[binary_index]
-    Reff2 = initconditions.set_reff(mass2_binary, metallicity=met, **kwargs)
-
-    # set radial velocity, set transverse velocity, set location in cluster
-    vr, vt, r = initconditions.set_vr_vt_r(N=mass1.size, **kwargs)
-
-    # set singles id
-    single_ids = np.arange(mass1.size)
-    binary_secondary_object_id = np.arange(mass1.size, mass1.size + mass2_binaries.size)
-
-    # set binind and correct masses of binaries
-    binind = np.zeros(mass1.size)
-    binind[binary_index] = np.arange(len(binary_index)) + 1
-    mass1[binary_index] += mass2_binary
+    Reff2 = initconditions.set_reff(mass2_binaries, metallicity=met, **kwargs)
+    ## TODO: CARL!!! MAKE GODDAMN SURE THESE ARE ALL IN RSUN AND NOT AU!!!!!
```

Comment:
Has this been made sure?

Reply:
I think that We covered this today and we await Carls final tweaks!